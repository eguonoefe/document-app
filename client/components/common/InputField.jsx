import React from 'react';
import PropTypes from 'prop-types';
/**
 * Input component that displays an Input Field.
 * @param {Object} props The props for the component.
 * @returns {String} The HTML markup for the Input component
 */
const InputField =
  ({ className, name, value, placeholder, type, onChange, label, id }) => (
    <div className="input-field">
      <input
        className={className}
        id={id}
        type={type}
        name={name}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        required
      />
      <label
        className="form-label"
        htmlFor="email" data-error="Invalid Input" data-success="">
        {label}
      </label>
    </div>
  );

// Set Input PropTypes
InputField.propTypes = {
  className: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.node,
  placeholder: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  label: PropTypes.string,
  id: PropTypes.string
};

export default InputField;

