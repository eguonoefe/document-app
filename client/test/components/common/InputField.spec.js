/* global expect jest test */
import React from 'react';
import { shallow,configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import InputField from '../../../components/common/InputField';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/common/InputField');

describe('InputField Component', () => {
  test('should match the inputfield snapshot', () => {
    const onChange = jest.fn();
    const component = shallow(
      <InputField
        onChange={onChange}
        className={'awesome'}
        name={'input'}
        value={'efe'}
        placeholder={'inputfield'}
        type={'text'}
        label={'input'} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
