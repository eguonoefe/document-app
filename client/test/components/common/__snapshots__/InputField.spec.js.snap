// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component should match the inputfield snapshot 1`] = `
<div
  className="input-field"
>
  <input
    className="awesome"
    name="input"
    onChange={[MockFunction]}
    placeholder="inputfield"
    required={true}
    type="text"
    value="efe"
  />
  <label
    className="form-label"
    data-error="Invalid Input"
    data-success=""
    htmlFor="email"
  >
    input
  </label>
</div>
`;
